src/logic/safe/store/actions/fetchSafe.ts

```diff
@@ -73,6 +79,29 @@ export const fetchSafe =
     // remote (client-gateway)
     if (remoteSafeInfo) {
       safeInfo = await extractRemoteSafeInfo(remoteSafeInfo)
+
+      const state = store.getState()
+
+      // If these polling timestamps have changed, fetch again
+      const { collectiblesTag, txQueuedTag, txHistoryTag } = currentSafe(state)
+
+      const isCollectiblesPage = !!matchPath<{ safeAddress: string }>(state.router.location.pathname, {
+        path: SAFE_ROUTES.ASSETS_COLLECTIBLES,
+      })
+
+      const shouldUpdateCollectibles = collectiblesTag !== safeInfo.collectiblesTag && isCollectiblesPage
+      const shouldUpdateTxHistory = txHistoryTag !== safeInfo.txHistoryTag
+      const shouldUpdateTxQueued = txQueuedTag !== safeInfo.txQueuedTag
+
+      if (shouldUpdateCollectibles) {
+        dispatch(fetchCollectibles(safeAddress))
+      }
+
+      if (shouldUpdateTxHistory || shouldUpdateTxQueued) {
+        dispatch(fetchTransactions(safeAddress))
+      }
+
+      dispatch(fetchSafeTokens(address))
```

Comment:
`fetchSafeTokens` doesn't depend on safe info, so it could be run in parallel with it, like before.